tests/gcp/operators/test_bigquery.py

```diff
@@ -432,6 +433,93 @@ def test_bigquery_operator_defaults(self, mock_hook):
         ti.render_templates()
         self.assertTrue(isinstance(ti.task.sql, str))
 
+    def test_bigquery_operator_extra_serialized_field_when_single_query(self):
+        with self.dag:
+            BigQueryOperator(
+                task_id=TASK_ID,
+                sql='SELECT * FROM test_table',
+            )
+        # ToDo: Make OperatorLinks working for list of sql queries
+        serialized_dag = SerializedDAG.to_dict(self.dag)
+        self.assertIn("sql", serialized_dag["dag"]["tasks"][0])
+
+        dag = SerializedDAG.from_dict(serialized_dag)
+        simple_task = dag.task_dict[TASK_ID]
+        self.assertEqual(getattr(simple_task, "sql"), 'SELECT * FROM test_table')
+
+        #########################################################
+        # Verify Operator Links work with Serialized Operator
+        #########################################################
+
+        # Check Serialized version of operator link
+        self.assertEqual(
+            serialized_dag["dag"]["tasks"][0]["_operator_extra_links"],
+            [{'airflow.gcp.operators.bigquery.BigQueryConsoleLink': {}}]
+        )
+
+        # Check DeSerialized version of operator link
+        self.assertIsInstance(list(simple_task.operator_extra_links)[0], BigQueryConsoleLink)
+
+        ti = TaskInstance(task=simple_task, execution_date=DEFAULT_DATE)
+        ti.xcom_push('job_id', 12345)
+
+        # check for positive case
+        url = simple_task.get_extra_links(DEFAULT_DATE, BigQueryConsoleLink.name)
+        self.assertEqual(url, 'https://console.cloud.google.com/bigquery?j=12345')
+
+        # check for negative case
+        url2 = simple_task.get_extra_links(datetime(2017, 1, 2), BigQueryConsoleLink.name)
+        self.assertEqual(url2, '')
+
+    def test_bigquery_operator_extra_serialized_field_when_multiple_queries(self):
+        with self.dag:
+            BigQueryOperator(
+                task_id=TASK_ID,
+                sql=['SELECT * FROM test_table', 'SELECT * FROM test_table2'],
+            )
+        serialized_dag = SerializedDAG.to_dict(self.dag)
+        self.assertIn("sql", serialized_dag["dag"]["tasks"][0])
+
+        dag = SerializedDAG.from_dict(serialized_dag)
+        simple_task = dag.task_dict[TASK_ID]
```

Comment:
```suggestion
        deserialized_task = dag.task_dict[TASK_ID]
```

As we've currently got something else called SimpleTask and this isn't one of those.

Reply:
It is just a task_id in tests specifically for Serialised Tests :D , does it matter? Happy to change the name but there are several places where the task_id is named simple_task. 

And I think we have SimpleTaskInstance and SimpleDagBag (that are soon to be removed too)